### ISSUE TITLE
Bump BoringSSL version to `fips-20220613` to match upstream Go

### ISSUE THREAD

**Issue by reedloden** [cloudflare/boring#203]
Upstream Go is [bumping the BoringSSL FIPS version](https://github.com/golang/go/issues/64717) used for BoringCrypto in order to pull in support for TLS 1.3 prior to the [NIST SP 800-52 Rev. 2](https://csrc.nist.gov/pubs/sp/800/52/r2/final) deadline of January 1, 2024. This updated version is still [in-process](https://csrc.nist.gov/Projects/cryptographic-module-validation-program/modules-in-process/modules-in-process-list) with NIST CMVP for FIPS 140-3 ("In Review" as of 2023-11-20).

Filing this as a tracking issue for doing the same thing in boring once this has officially landed and been released upstream. Last bump was done in #97.

**Comment by howardjohn** [comment]
FYI This was reverted in https://github.com/golang/go/issues/65321

**Comment by reedloden** [comment]
> FYI This was reverted in https://github.com/golang/go/issues/65321

Yeah, pending NIST CMVP certification. Hopefully, that will be soon. 🤞

**Comment by howardjohn** [comment]
The CMVP is ready now: https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4735
